Makefile

```
@@ -83,16 +85,21 @@ MSVC_PREFIX=LIB="$(MSVC_LIB)" INCLUDE="$(MSVC_INCLUDE)"
 MSVC64_LIB = $(MSVC_LIB1)/Lib/amd64;$(MSVC_LIB2)/Lib/x64
 MSVC64_PREFIX=LIB="$(MSVC64_LIB)" INCLUDE="$(MSVC_INCLUDE)"
 
-MSVCC = $(MSVCC_ROOT)/cl.exe $(MSVC_FLAGS)
-MSVCC64 = $(MSVCC_ROOT)/amd64/cl.exe $(MSVC_FLAGS)
+MSVCC = $(MSVCC_ROOT)/cl.exe $(MSVC_CFLAGS)
+MSVCC64 = $(MSVCC_ROOT)/amd64/cl.exe $(MSVC_CFLAGS)
 else
 MSVCC_ROOT:=
 MSVC_PREFIX=PATH="$(MSVS_PATH)$(PATH)" LIB="$(MSVS_LIB)$(LIB)" INCLUDE="$(MSVS_INC)$(INCLUDE)"
 MSVC64_PREFIX=PATH="$(MSVS64_PATH)$(PATH)" LIB="$(MSVS64_LIB)$(LIB)" INCLUDE="$(MSVS64_INC)$(INCLUDE)"
 
-MSVCC = cl.exe $(MSVC_FLAGS)
-MSVCC64 = cl.exe $(MSVC_FLAGS)
+MSVCC = cl.exe $(MSVC_CFLAGS)
+MSVCC64 = cl.exe $(MSVC_CFLAGS)
+endif
 endif
+
+ifneq "$(filter True true,$(CI))" ""
```

I didn’t think of this previously - unlike where we were using `GITHUB_ACTIONS` in ocaml/ocaml#12842, but we need to be sure that this can only be triggered by a workflow/CI run _here_ - at the moment, I think this will enable warnings as errors on all GitHub Actions runs building the compiler?

I think therefore should be a variable which is set by the AppVeyor script e.g. adding `FLEXDLL_WARN_ERROR: true` to the `global` section of `environment` in appveyor.yml and then having:

```suggestion
FLEXDLL_WARN_ERROR ?=
ifneq "$(filter True true,$(FLEXDLL_WARN_ERROR))" ""
```